ci/da-ghc-lib/compile.yml

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+parameters:
+- name: final_job_name
+
+jobs:
+
+- job: ${{ parameters.final_job_name }}
+  pool:
+    vmImage: 'Ubuntu 16.04'
+  variables:
+    base-sha: '9c787d4d24f2b515934c8503ee2bbd7cfac4da20'
+    patches: '4650347db9a7e8c65831ab83b80da43b28c3deba 833ca63be2ab14871874ccb697'
+    ghc-lib-sha: '952911541233fa8a5bd94486bc46e03688b23bdd'
+    flavor: 'ghc-8.8.1'
+  steps:
+  - checkout: self
+  - bash: |
+      set -euo pipefail
+
+      CACHE_KEY="$(find ci/da-ghc-lib -type f -print0 | sort -z | xargs -r0 md5sum | md5sum | awk '{print $1}')"
+      LIB="da-ghc-lib/ghc-lib-$CACHE_KEY.tar.gz"
+      PARSER="da-ghc-lib/ghc-lib-parser-$CACHE_KEY.tar.gz"
+      URL_BASE="https://daml-binaries.da-ext.net"
+      if [ "200" = "$(curl -Is "$URL_BASE/$LIB" | head -1 | awk '{print $2}')" ] \
+        && [ "200" = "$(curl -Is "$URL_BASE/$PARSER" | head -1 | awk '{print $2}')" ]; then
```

Comment:
Might be worth extracting the calls to `curl | head | awk`. If either of those fails due to a network connectivity issue (as opposed to printing "404"), I think we'd probably want to abort.

Reply:
Done, I think. Is this what you had in mind? Cursory testing in a shell does seem to indicate this would fail as expected, whereas the previous version wouldn't, as you suggested.

I've also added a `-n` to the `gsutil cp` commands so that even if we do end up running this when not needed, we would not overwrite the files.